stock_picking_return_refund_option/models/stock_picking.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    to_refund_lines = fields.Selection(selection=[
+        ('keep_line_value', 'Keep Line Value'),
+        ('to_refund', 'To Refund'),
+        ('no_refund', 'No Refund')],
+        string='Refund Options', default='keep_line_value',
+        help="This field allow modify 'to_refund' field value in all "
+             "stock moves from this picking after it has been confirmed."
+             "keep_line_value: keep the original value.\n"
+             "to_refund: set all stock moves to True value and recompute "
+             "delivered quantities in sale order.\n"
+             "no_refund: set all stock moves to False value and recompute "
+             "delivered quantities in sale/purchase order.",
+    )
+    is_return = fields.Boolean(compute='_compute_is_return')
+
+    def _compute_is_return(self):
+        for picking in self:
+            if any(x.origin_returned_move_id for x in picking.move_lines):
+                picking.is_return = True
+
+    @api.multi
+    def _update_stock_moves(self):
+        for pick in self.filtered(
+                lambda x: x.to_refund_lines != 'keep_line_value'):
+            pick.move_lines.write({
+                'to_refund': pick.to_refund_lines == 'to_refund',
+            })
+        for move in self.filtered(
+                lambda x: x.to_refund_lines == 'keep_line_value'
+                ).mapped('move_lines'):
+            move.to_refund = move.origin_to_refund
+
+    @api.multi
+    def set_delivered_qty(self):
+        """
+        Check if exists sale_line_id field in stock.move model that has been
+        added by sale_stock module, this module has not dependency of this,
+        Update sale order line qty_delivered for allow do a refund invoice
+        """
+        if hasattr(self.env['stock.move'], 'sale_line_id') and self.sale_id:
```

Comment:
In the init hook you install sale_stock. Here you write code checking if the module is installed. If your code work in a different way if the module `sale_stock` is installed and the module doens't depend of `sale_stock` you must define a glue addon with the specific code.

Reply:
Sometimes you need only the sales part, sometimes only the purchase part, and sometimes both. With this you avoid 2 ugly glue modules and perform the same without problems.

Reply:
@pedrobaeza It's not a good practice at all. We must ensure a proper separation between modules and this kind of code is not sustainable.

Reply:
I don't agree. I prefer this option. This technique avoids a lot of module overhead and it's sustainable through the trick in testing (we are testing all the code). In upper versions, if for example a field name changes, we will detect it the same as having 3 modules.

We have spent time assuring that all code is tested and thus the `_install_module_hook` trick. You can't dismiss this solution if it's only a matter of choice. Technically, it has no problem. Functionally neither.

Reply:
Conceptually it has.